### PR TITLE
feat: save stability artifacts to DIAL file storage

### LINKED ISSUE #29: Switch on using file API for Stable-diffusion instead of embedding image in the conversation
See https://github.com/epam/ai-dial-core/issues/29 for details

### PR DESCRIPTION
**Pull request by adubovik**
Fixes #29

### CODE REVIEW

**Comment by adubovik** [comment]
/deploy-review
>Deployment status: success

**Comment by ai-dial-actions** [comment]
/deploy-review
>Deployment status: success

File: aidial_adapter_bedrock/llm/model/stability.py [hunk 1]
```diff
@@ -73,14 +71,44 @@ def prepare_input(prompt: str) -> Dict[str, Any]:
     return {"text_prompts": [{"text": prompt}]}
 
 
+async def save_to_storage(
+    storage: ImageStorage, attachment: Attachment
+) -> Attachment:
+    if attachment.type == "image/png" and attachment.data is not None:
```
Comment (Oleksii-Klimov) [inline aidial_adapter_bedrock/llm/model/stability.py]: Is this check really needed? I think the ".png" extension in the url is redundant.

File: aidial_adapter_bedrock/dial_api/storage.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        self.storage = FileStorage(dial_url, base_dir, api_key)
+
+    @staticmethod
+    def hash_digest(string: str) -> str:
+        return hashlib.sha256(string.encode()).hexdigest()
+
+    async def upload_base64_png_image(self, data: str) -> FileMetadata:
+        filename = ImageStorage.hash_digest(data) + ".png"
+        content: bytes = base64.b64decode(data)
+        return await self.storage.upload(filename, "image/png", content)
```
Comment (Oleksii-Klimov) [inline aidial_adapter_bedrock/dial_api/storage.py]: Why is the attachment type not passed as an argument?

**Comment by adubovik** [comment]
/deploy-review
>Deployment status: success

**Comment by adubovik** [comment]
/deploy-review
>Deployment status: success

**Comment by adubovik** [comment]
/deploy-review
>Deployment status: success

**Review by Oleksii-Klimov — APPROVED** [review]

File: aidial_adapter_bedrock/llm/model/stability.py (hunk 1 above)
Comment (adubovik) [reply aidial_adapter_bedrock/llm/model/stability.py]: Fixed. Removed the check.

File: aidial_adapter_bedrock/dial_api/storage.py (hunk 2 above)
Comment (adubovik) [reply aidial_adapter_bedrock/dial_api/storage.py]: Fixed

**Pull request merged**